src/term/transform/linearize.rs

```diff
@@ -86,8 +87,14 @@ fn term_to_affine(term: &mut Term, var_uses: &mut HashMap<Name, Val>, let_bodies
     // Var-declaring terms
     Term::Lam { nam, bod } => {
       if let Some(nam_some) = nam {
-        if let Some(uses) = var_uses.get(nam_some).copied() {
+        if let Some(mut uses) = var_uses.get(nam_some).copied() {
           term_to_affine(bod, var_uses, let_bodies);
+
+          if uses != 0 {
+            let name = Name(format!("{nam_some}$used"));
```

Comment:
Can you refactor this into a function or a lambda so that there's only one place deciding what the name of this generated variable is?